lib/components/TextField/index.js

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import styled, { css } from 'styled-components';
+import theme from '../theme';
+
+import InputField from '../InputField';
+import InputFieldError from '../InputFieldError';
+import InputFieldLabel from '../InputFieldLabel';
+
+const TextField = ({
+  className,
+  disabled,
+  id,
+  error,
+  label,
+  required,
+  ...props
+}) => {
+  const cx = classNames(className);
+
+  return (
+    <InputField className={cx}>
+      {label && <InputFieldLabel htmlFor={id}>{label}</InputFieldLabel>}
```

Comment:
please-comment: I see you're not using the nesting pattern of putting controls inside labels in this case, and I'm guessing that's to allow the possibility that we may want to render this without a label. Why is that important here, but not in the other cases? Worth a comment.

Reply:
Trying to find a balance between getting this done and making things perfect, so I partly copied this from triton.

In triton, I believe we have some fields that we don't use `label`s on. I'd like to move away from this, since we then need to handle accessibility in some other manner, and I'd like to try reduce the custom UI unless we have a really good reason to.

So I just slapped a `label: PropTypes.any.isRequired` and called it good for now. I didn't want want to spend the time trying to think through all the possible future scenarios, and if we want to commit to requiring it now when we are using some fields without it.

For example, would it be beneficial to adapt this InputField component to wrap all components, even checkboxes, so we can use it to have a common error message style? In that case, checkboxes have a different label style. So that does that mean all field labels are handled outside of InputField? What about calendars? Do we ever want checkboxes to have their own error? If not, then wrapping it in InputField doesn't make sense. But there could be a case we would want it, like just a toggle for "turn this feature on", so maybe we do? Some of this stuff seemed like optimizations that could be made when/if we get to using perts-ui in Copilot and we start to see more varied use.

Being in get it done mode, it seemed like not prematurely optimizing and rewriting the component and adjusting the styles to allow for `<label><input></label>` was the best path.

Let me know what you think.

<img width="862" alt="Screen Shot 2020-05-18 at 3 57 42 PM" src="https://user-images.githubusercontent.com/276924/82267115-56a6da00-9920-11ea-8349-0ed2205ca776.png">

<img width="861" alt="Screen Shot 2020-05-18 at 3 56 01 PM" src="https://user-images.githubusercontent.com/276924/82267119-59093400-9920-11ea-8c33-3c34aec30c55.png">

Reply:
I see the dilemma, thanks.